### DISCUSSION TITLE
Restricted Inverse Wishart Distribution

### DISCUSSION THREAD

**Discussion by ghost** [pymc-devs/pymc3#5396] [Ideas]
While the inverse Wishart distribution has significant problems, it also has advantages:
1. Conjugacy.
2. A strong maximum entropy argument (the inverse Wishart is the least informative prior for a given harmonic and geometric mean). 
3. The Wishart and inverse-Wishart distributions can be used as _informative_ priors, unlike the LKJ prior on correlations, which is a purely regularizing prior. While a purely regularizing prior that shrinks correlations towards 0 is a good idea in cases where the likelihood swamps the prior, in cases where there is important prior information that should be included in the model about the sign and size of correlations, this is not optimal.
In general, there are two strong arguments against using the inverse Wishart:
1. The inverse Wishart is highly informative, even when its parameters are set to very small values. This is correct, but doesn't imply they can't be useful when prior information suggests an inverse-Wishart is reasonable, e.g. when using the sampling distribution from a past study.
2. The inverse Wishart is difficult to interpret and unsuited to HMC. While this is correct in its usual form, it is possible to reparametrize the distribution into the restricted inverse-Wishart distribution using the separation strategy. Such distributions are easier to sample from and to interpret -- see [here](https://arxiv.org/pdf/1809.04746.pdf) and [here](http://www3.stat.sinica.edu.tw/statistica/oldpdf/A10n416.pdf).
Given the restricted inverse Wishart retains some of the advantages of its more common Wishart counterpart while avoiding some of its pitfalls, I think it makes sense to include it in PyMC3 for modeling informative priors, as long as a note is included specifying that it is not a good uninformative prior.

**Comment by ckrapu** [comment]
The first paper you linked asserts that the LKJ and restricted inverse Wishart distributions are equivalent. What would be the point of adding in the RIW distribution? Would an additional parameterization for the LKJ suffice?

**Comment by ghost** [comment]
> The first paper you linked asserts that the LKJ and restricted inverse Wishart distributions are equivalent. What would be the point of adding in the RIW distribution? Would an additional parameterization for the LKJ suffice?

The LKJ is equivalent to the restricted Wishart, not restricted inverse Wishart.

**Comment by ckrapu** [comment]
Ah, okay. My mistake.
